out_writeapi_test.go

```diff
@@ -38,3 +47,244 @@ func TestFLBPluginRegister(t *testing.T) {
+				return nil, fmt.Errorf("Failed to unmarshal")
+			}
+
+			textField := message.Get(md.Fields().ByJSONName("Text"))
+			timeField := message.Get(md.Fields().ByJSONName("Time"))
+
+			if textField.String() != "FOO" && textField.String() != "BAR" {
+				return nil, fmt.Errorf("Wrong Text Value: %s", textField.String())
+			}
+			log.Printf("String: %s", textField.String())
+			if timeField.String() != "000" {
+				return nil, fmt.Errorf("Wrong Time Value")
+			}
+
+			res = append(res, true)
+			res = append(res, true)
+			return nil, nil
+		},
+		FinalizeFunc: func(ctx context.Context, opts ...gax.CallOption) (int64, error) {
+			return 0, nil
+		},
+		FlushRowsFunc: func(ctx context.Context, offset int64, opts ...gax.CallOption) (int64, error) {
+			return 0, nil
+		},
+		StreamNameFunc: func() string {
+			return ""
+		},
+	}
+
+	origFunc := getWriter
+	getWriter = func(client ManagedWriterClient, ctx context.Context, projectID string, opts ...managedwriter.WriterOption) (MWManagedStream, error) {
+		return mockMS, nil
+	}
+	defer func() { getWriter = origFunc }()
+
+	patch1 := monkey.Patch(output.FLBPluginConfigKey, func(plugin unsafe.Pointer, key string) string {
+		return ""
+	})
+	defer patch1.Unpatch()
+
+	patchSetContext := monkey.Patch(output.FLBPluginSetContext, func(plugin unsafe.Pointer, ctx interface{}) {
+		setID = ctx.(int)
+	})
+	defer patchSetContext.Unpatch()
+
+	plugin := unsafe.Pointer(nil)
+	initRes := FLBPluginInit(plugin)
+
+	orgFunc := getFLBPluginContext
+	getFLBPluginContext = func(ctx unsafe.Pointer) int {
+		checks.calledGetContext++
+		return setID
+	}
+	defer func() { getFLBPluginContext = orgFunc }()
+
+	origReadyFunc := isReady
+	isReady = func(queueHead *managedwriter.AppendResult) bool {
+		if res[0] {
+			return true
+		} else {
+			return false
+		}
+	}
+	defer func() { isReady = origReadyFunc }()
+
+	origResultFunc := pluginGetResult
+	pluginGetResult = func(queueHead *managedwriter.AppendResult, ctx context.Context) (int64, error) {
+		if res[1] {
+			checks.appendQueue++
+			return -1, nil
+		}
+		err := errors.New("Failed to Get Result")
+		return 0, err
+	}
+	defer func() { pluginGetResult = origResultFunc }()
+
+	patchDecoder := monkey.Patch(output.NewDecoder, func(data unsafe.Pointer, length int) *output.FLBDecoder {
+		checks.createDecoder++
+		return nil
+	})
+	defer patchDecoder.Unpatch()
+
+	var loopCount int = 0
+	patchRecord := monkey.Patch(output.GetRecord, func(dec *output.FLBDecoder) (ret int, ts interface{}, rec map[interface{}]interface{}) {
+		checks.gotRecord++
+		dummyRecord := make(map[interface{}]interface{})
+		if loopCount%2 == 0 {
+			loopCount++
+			dummyRecord["Text"] = []byte{70, 79, 79}
+			dummyRecord["Time"] = []byte{48, 48, 48}
+			return 0, nil, dummyRecord
+		} else {
+			loopCount++
+			return 1, nil, dummyRecord
+		}
+	})
+	defer patchRecord.Unpatch()
+
+	origRequestFunc := sendRequest
+	sendRequest = func(ctx context.Context, data [][]byte, config **outputConfig) error {
```

Comment:
Instead of mocking sendRequest you can intercept the call to AppendRows, right?

Reply:
Done!